core/iwasm/fast-jit/fe/jit_emit_numberic.c

```diff
@@ -472,16 +472,104 @@ jit_compile_op_i64_bitwise(JitCompContext *cc, IntBitwise bitwise_op)
     return false;
 }
 
+static bool
+compile_int_shift(JitCompContext *cc, IntShift shift_op, bool is_i32)
+{
+#if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
+    JitReg ecx_hreg = jit_codegen_get_hreg_by_name("ecx");
+    JitReg rcx_hreg = jit_codegen_get_hreg_by_name("rcx");
+#endif
+    JitReg left, right, mod_right, res;
+
+    POP_INT(right);
+    POP_INT(left);
+
+    if (IS_CONST_ZERO(right)) {
+        res = left;
+        goto shortcut;
+    }
+
+    /* right modulo N */
+    if (is_i32) {
+        mod_right = jit_cc_new_reg_I32(cc);
+        GEN_INSN(REM_U, mod_right, right, NEW_CONST(I32, 32));
+    }
+    else {
+        mod_right = jit_cc_new_reg_I64(cc);
+        GEN_INSN(REM_U, mod_right, right, NEW_CONST(I64, 64));
+    }
+
+    if (IS_CONST_ZERO(right)) {
+        res = left;
```

Comment:
Code is duplicated with L487-489, do you want to check `IS_CONST_ZERO(mod_right)`?
But it is not const reg. How about change L487-489 to:
```
if (jit_reg_is_const_val(right)) {
    if ((is_i32 && jit_cc_get_const_I32(cc, right) % 32 == 0))
        || (!is_i32 && jit_cc_get_const_I64(cc, right) % 64 == 0)) {
        res = left;
        goto shortcut;
      }
  }
```